carrot-log4j-aws-sns/src/main/java/com/carrotgarden/log4j/aws/sns/Appender.java

```diff
@@ -203,8 +209,11 @@ public boolean hasAmazonClient() {
 
 	/** provide amazon login credentials from file */
 	protected boolean ensureCredentials() {
-
-        if (hasCredentials()) {
+		if (hasAwsCredentialsOverride()) {
+			amazonCredentials = new AWSCredentialsProviderChain(new InstanceProfileCredentialsProvider(),
+																new EnvironmentVariableCredentialsProvider(),
```

Comment:
Oh, I think I follow you - you're saying just make this part of either chain (regardless of whether we're also adding a url/file properties provider at the end). I thought you were saying we would just create a single chain (in which case I was worried PropertiesFileCredentialsProvider would throw an exception because the file may not exist).

I can update it to remove the override flag - that makes sense